### DISCUSSION TITLE
Update 2.3.3 broke my libraries and mess up with colours

### DISCUSSION THREAD

**Discussion by ZeoSonE** [rsayfutdinov/Pikka#16] [Q&A]
Hi,
Last update done a lot of wrong things with my libraries and colours.
1. I've completely lost my 2 libraries 
2. Libraries changed the order and colour code ("History was always first and all libraries has a different colour tag)
<img width="374" alt="Screenshot 2021-07-07 at 22 38 57" src="https://user-images.githubusercontent.com/7634694/124832362-79f17600-df74-11eb-9f86-27f1221029de.png">
3. In the menu now I have many duplicate colours
<img width="287" alt="Screenshot 2021-07-07 at 22 38 33" src="https://user-images.githubusercontent.com/7634694/124832467-a0afac80-df74-11eb-8646-fe587fe1e864.png">

Please advise

**Comment by ZeoSonE** [comment]
Some more.
Every time I've restarted Pikka look what happened with the menu colours

<img width="287" alt="Screenshot 2021-07-07 at 22 44 45" src="https://user-images.githubusercontent.com/7634694/124832784-19166d80-df75-11eb-94a1-853bae1aebba.png">
<img width="287" alt="Screenshot 2021-07-07 at 22 45 01" src="https://user-images.githubusercontent.com/7634694/124832789-1b78c780-df75-11eb-8e21-31c39f32c084.png">

**Reply by rsayfutdinov** [reply]
> Some more.
> Every time I've restarted Pikka look what happened with the menu colours
> 
> <img alt="Screenshot 2021-07-07 at 22 44 45" width="287" src="https://user-images.githubusercontent.com/7634694/124832784-19166d80-df75-11eb-94a1-853bae1aebba.png">
> 
> <img alt="Screenshot 2021-07-07 at 22 45 01" width="287" src="https://user-images.githubusercontent.com/7634694/124832789-1b78c780-df75-11eb-8e21-31c39f32c084.png">

Hi, 

Would you try one thing for me?
Please, disable iCloud sync in Settings and restart the app couple of times. Is the issue with duplicates still there?

**Comment by rsayfutdinov** [comment]
> Hi,
> Last update done a lot of wrong things with my libraries and colours.
> 
> 1. I've completely lost my 2 libraries
> 2. Libraries changed the order and colour code ("History was always first and all libraries has a different colour tag)
> 
> <img alt="Screenshot 2021-07-07 at 22 38 57" width="374" src="https://user-images.githubusercontent.com/7634694/124832362-79f17600-df74-11eb-9f86-27f1221029de.png">
> 
> 3. In the menu now I have many duplicate colours
> <img alt="Screenshot 2021-07-07 at 22 38 33" width="287" src="https://user-images.githubusercontent.com/7634694/124832467-a0afac80-df74-11eb-8646-fe587fe1e864.png">
> 
> Please advise

Hi,

I have pushed an update to the app store. 
Please try to use "Settings->Migrate" button and then restart the app.
Let me know if it fixes the issue.

PS: Sorry that you have these issues. I really needed to migrate the application database, to be able to improve the overall performance and implement new features.

Regrads,
Ruslan.

**Comment by ZeoSonE** [comment]
Hi there

Please see screen recording related to the issue

https://a1d.d.pr/MAb4aW <https://a1d.d.pr/MAb4aW>


Cheers

On 8 Jul 2021, at 07:21, ruslan ***@***.***> wrote:


Some more.
Every time I've restarted Pikka look what happened with the menu colours

 <https://user-images.githubusercontent.com/7634694/124832784-19166d80-df75-11eb-94a1-853bae1aebba.png>  <https://user-images.githubusercontent.com/7634694/124832789-1b78c780-df75-11eb-8e21-31c39f32c084.png>
Hi,

Would you try one thing for me?
Please, disable iCloud sync in Settings and restart the app couple of times. Is the issue with duplicates still there?

**Comment by ZeoSonE** [comment]
Is this “Migrate” thing will be available in the update?

Currently can’t see any of these venues in the App

Thanks

On 8 Jul 2021, at 08:50, ruslan ***@***.***> wrote:

Please try to use "Settings->Migrate" button and then restart the app.

**Reply by ZeoSonE** [reply]
Ruslan, I'm currently working on the project and just copied a 6 digit code (310928) from the table. It wasn't a colour code, just a numeric code
Pikka instantly added two(!) hex value into the history and into the menu

**Reply by rsayfutdinov** [reply]
> Is this “Migrate” thing will be available in the update? Currently can’t see any of these venues in the App Thanks On 8 Jul 2021, at 08:50, ruslan ***@***.***> wrote: Please try to use "Settings->Migrate" button and then restart the app.

Yes. Will be available with the update. I hope it comes quickly through the apple review.

**Reply by rsayfutdinov** [reply]
> Ruslan, I'm currently working on the project and just copied a 6 digit code (310928) from the table. It wasn't a colour code, just a numeric code
> Pikka instantly added two(!) hex value into the history and into the menu

Yep. Got this.

Btw. I really appreciate your help! 🙏

**Reply by maskinmedia** [reply]
resetting worked. Restored my library but - a whoooole bunch of new folder that also appeered with the new version has to be manually deleted.

**Comment by ZeoSonE** [comment]
You are very welcome



On 8 Jul 2021, at 09:28, ruslan ***@***.***> wrote:


Ruslan, I'm currently working on the project and just copied a 6 digit code (310928) from the table. It wasn't a colour code, just a numeric code
Pikka instantly added two(!) hex value into the history and into the menu

Yep. Got this.

Btw. I really appreciate your help! 🙏

**Comment by ZeoSonE** [comment]
Ruslan,

Would be nice to have a more flexibility on sections/categories, like reorder, more colour options…

Have a look how Paste app doing this:





Thanks

On 8 Jul 2021, at 09:28, ruslan ***@***.***> wrote:


Ruslan, I'm currently working on the project and just copied a 6 digit code (310928) from the table. It wasn't a colour code, just a numeric code
Pikka instantly added two(!) hex value into the history and into the menu

Yep. Got this.

Btw. I really appreciate your help! 🙏

**Comment by rsayfutdinov** [comment]
> Ruslan,
> 
> Would be nice to have a more flexibility on sections/categories, like reorder, more colour options…
> 
> Have a look how Paste app doing this:
> 
> 
> 
> 
> 
> Thanks
> 
> On 8 Jul 2021, at 09:28, ruslan ***@***.***> wrote:
> 
> 
> Ruslan, I'm currently working on the project and just copied a 6 digit code (310928) from the table. It wasn't a colour code, just a numeric code
> Pikka instantly added two(!) hex value into the history and into the menu
> 
> Yep. Got this.
> 
> Btw. I really appreciate your help! 🙏
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/rsayfutdinov/Pikka/discussions/16#discussioncomment-978361>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB2H6BQGSRJNVSMQZ4VXXULTWVOS7ANCNFSM477PIA3A>.
> 
> 
> 

I'm already working on the library update. If you want I can send you an early version as soon as I think is ready:) so you can provide feedback at an early stage.

**Comment by rsayfutdinov** [comment]
I got it. Duplicate values have nothing to do with the database migration and version 2.3.3. The issue is older. Every time you start the app, Pikka transforms the last copied value from the clipboard to hex color value and automaticaly append it to the history.

**Comment by ZeoSonE** [comment]
I would be agreed with that if it wouldn’t add double values on copy a number/code.

Cheers.

On 8 Jul 2021, at 13:59, ruslan ***@***.***> wrote:


I got it. Duplicate values have nothing to do with the database migration and version 2.3.3. The issue is older. Every time you start the app, Pikka transforms the last copied value from the clipboard to hex color value and automaticaly append it to the history.
